plugins/serializers/prometheusremotewrite/prometheusremotewrite.go

```diff
@@ -162,18 +177,24 @@ func (s *Serializer) SerializeBatch(metrics []telegraf.Metric) ([]byte, error) {
 			}
 
 			// A batch of metrics can contain multiple values for a single
-			// Prometheus sample.  If this metric is older than the existing
+			// Prometheus sample. If this metric is older than the existing
 			// sample then we can skip over it.
 			m, ok := entries[metrickey]
 			if ok {
 				if metric.Time().Before(time.Unix(0, m.Samples[0].Timestamp*1_000_000)) {
+					lastErr = fmt.Errorf("metric %q has samples with timestamp %v older than already registered before", metric.Name(), metric.Time())
 					continue
 				}
 			}
 			entries[metrickey] = promts
 		}
 	}
 
+	if lastErr != nil && s.Log != nil {
+		// log only the last recorded error, as it could be too verbose to log every one
+		s.Log.Errorf("some series were dropped, %d series left to send; last recorded error: %v", len(entries), lastErr)
+	}
```

Comment:
Please do not check for `s.Log` being `nil` as this cannot happen in usual operations. Fix the tests instead.

Reply:
Removed in https://github.com/influxdata/telegraf/pull/15893/commits/265065b915a2753a7597139d3886f14e570fd816